### PR TITLE
Setup Redux and Redux-Saga

### PR DESCRIPTION
**Pull request by alexmochu**
##### What does this PR do?
It sets up Redux which includes the store and integrates Redux-Saga middleware

##### Description of Task to be completed?

- Add redux, react-redux and react-saga libraries
- Setup the Redux store in relation to the re-ducks folder structure
- Integrate the Redux-Saga to the store

### CODE REVIEW

File: src/app/Home/components/HomeComponent.js [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+import PropTypes from 'prop-types';
 
-const HomeComponent = () => (
-  <h1 className='home__title'>Welcome to Andela Societies </h1>
-);
+import { connect } from 'react-redux';
 
-export default HomeComponent;
+export class App extends Component {
+  onIncrement = () => {
```
Comment (Kachulio1) [inline src/app/Home/components/HomeComponent.js]: you should move the actions to the ducks folder

File: src/app/Home/components/HomeComponent.js [hunk 2]
```diff
@@ -1,7 +1,67 @@
...
+      type: 'INCREMENT_ASYNC',
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h1 className='home__title'>Welcome to Andela Societies</h1>
+        <div className='home__container'>
+          <button type='button' onClick={this.onIncrement}>
```
Comment (Kachulio1) [inline src/app/Home/components/HomeComponent.js]: create a testSaga component or * and move this code there and just import the component in app index,

File: src/app/Home/components/HomeComponent.js [hunk 3]
```diff
@@ -1,7 +1,67 @@
...
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 
-const HomeComponent = () => (
-  <h1 className='home__title'>Welcome to Andela Societies </h1>
-);
+import { connect } from 'react-redux';
 
-export default HomeComponent;
+export class App extends Component {
```
Comment (Kachulio1) [inline src/app/Home/components/HomeComponent.js]: Don't forget to put this back to homecomponent

File: src/app/Home/components/HomeComponent.js (hunk 1 above)
Comment (alexmochu) [reply src/app/Home/components/HomeComponent.js]: @Kachulio1 Could you please expound more on what you mean by saying I should move the actions to the ducks folder.

File: package.json [hunk 4]
```diff
@@ -59,6 +59,10 @@
   "dependencies": {
     "react": "^16.6.3",
     "react-dom": "^16.6.3",
-    "react-router-dom": "^4.3.1"
+    "react-redux": "^6.0.0",
+    "react-router-dom": "^4.3.1",
+    "redux": "^4.0.1",
+    "redux-devtools-extension": "^2.13.7",
```
[the remark this answers is not in the document]
Comment (alexmochu) [reply package.json]: The above libraries when listed on 'devDependencies' are causing bundling errors that they should be listed on 'dependencies'.

File: src/app/Home/components/HomeComponent.js (hunk 3 above)
Comment (alexmochu) [reply src/app/Home/components/HomeComponent.js]: I have reverted to HomeComponent

File: src/app/Home/components/HomeComponent.js (hunk 2 above)
Comment (alexmochu) [reply src/app/Home/components/HomeComponent.js]: I have created a stateless testSaga component for this.

**Comment by Chris-Maina** [comment]
LGTM

File: src/index.js [hunk 5]
```diff
@@ -1,8 +1,17 @@
...
+import { Provider } from 'react-redux';
 
-import HomeComponent from './app/Home/components';
+import Router from './routes';
+import configureStore from './store';
 
 import './styles/main.scss';
 
-ReactDOM.render(<HomeComponent />, document.getElementById('root'));
+const store = configureStore();
```
Comment (Chris-Maina) [inline src/index.js]: @alexmochu `configureStore()` takes initial state. You have not provided that here

Comment (alexmochu) [reply src/index.js]: @Chris-Maina I have configured the Store here --> https://github.com/andela/andela-societies-frontend/pull/170/files#diff-14

This is where I have provided the initial state.

File: src/store.js [hunk 6]
```diff
@@ -1 +1,22 @@
 
+
```
Comment (Kachulio1) [inline src/store.js]: in jest, config ignore store and other .config files 
which are bringing  the coverage down

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexmochu) [reply package.json]: @Kachulio1  The babel stuff is already in devDependencies. Installing redux-devtools-extension in devDependencies is causing the following compilation error. It should be in dependencies.
screenshot 2019-02-06 at 12 26 25

[the remark this answers is not in the document]
Comment (Kachulio1) [reply package.json]: ok maybe keep 'the dev tool then

File: src/store.js (hunk 6 above)
Comment (alexmochu) [reply src/store.js]: @Kachulio1  I have ignored the files that are bringing the coverage down.

**Pull request merged**